Sources/Client/Player.cpp

```diff
@@ -788,10 +788,16 @@ namespace spades {
 
 			horzModifier *= sqrt(1 - pow(o.z, 4));
 
-			o += GetRight() * rec.x * triWave * horzModifier;
-			o += GetUp() * std::min(rec.y, std::max(0.f, upLimit)) * vertModifier;
-			o = o.Normalize();
-			SetOrientation(o);
+			//If the client player is spectator disable recoil
```

Comment:
I think the correct distinction is whether it's a local player (for which we must apply the recoil) or a non-local player (for which we mustn't because the recoil is applied by their client, and we are supposed to just display the result). You can check that by `this->IsLocalPlayer()`.

The current code exhibits a different behavior depending on whether you are a spectator or dead and temporarily observing a teammate, which isn't probably what we want.

Reply:
Good point. Will force push the corrected code once im home.